content/300-guides/300-upgrade-guides/300-upgrading-to-use-preview-features/050-enabling-named-constraints.mdx

```diff
@@ -118,121 +113,89 @@ Use the following guide if your project uses Prisma Migrate. Choose the upgrade
+   This migration renames any constraints that do not currently follow Prisma's naming convention.
 
-    ```terminal
-    npx prisma migrate dev
-    ```
+1. Run the `prisma migrate deploy` command to apply the migration to your production environment:
 
-1.  Deploy the migration to production and test environments.
+   ```terminal
+   npx prisma deploy
+   ```
 
-    ```terminal
-    npx prisma migrate deploy
-    ```
+## Checking whether your environments use identical names
 
-### Upgrade path 2: I want to use Prismas default constraint and index names
+reiterate why this might be the case
 
-1. Enable the `namedConstraints` Preview:
+In order to detect this, backup your current prisma.schema and then run prisma db pull against both environments saving the results to their own files.
+Then diff both files. If you see differences in constraint names, your production and local environments are out of sync.
 
-   ```prisma highlight=3;normal
-   generator client {
-     provider        = "prisma-client-js"
-     previewFeatures = ["namedConstraints"]
-   }
-   ```
+In the following example, the `Post` model has a foreign key constraint with a custom name in production that does not match development:
 
-1. Run the `prisma migrate dev` command to generate a new migration:
+  <TabbedContent tabs={[<FileWithIcon text="Development environment" icon="prisma"/>, <FileWithIcon text="Production environment" icon="prisma"/>]}>
 
-   ```terminal
-   npx prisma migrate dev
-   ```
+  <tab>
 
-   This migration renames any constraints that do not currently follow Prisma's naming convention.
+```prisma highlight=5;normal
+model Post {
+  id    Int @id @default(autoincrement())
+  title String
+  authorName String @default("Anonymous")
+  author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+}
+```
 
-## Upgrade path for introspection only users
+  </tab>
 
-Use the following guide if you use introspection only.
+  <tab>
 
-1. Enable the `namedConstraints` Preview:
+```prisma highlight=5;normal
+model Post {
+  id    Int @id @default(autoincrement())
+  title String
+  authorName String @default("Anonymous")
+  author User? @relation(fields: [authorName], references: [name], map: "Custom_Production_Name")
+}
+```
 
-   ```prisma highlight=3;normal
-   generator client {
-     provider        = "prisma-client-js"
-     previewFeatures = ["namedConstraints"]
-   }
-   ```
+  </tab>
 
-1. Introspect to populate your Prisma schema with any constraint names that do not follow Prisma's naming convention:
+  </TabbedContent>
 
-   ```terminal
-   npx prisma introspect
-   ```
+## Aligning your environments if their constraint or index names differ
 
-   For example, in the following model, the `@id` primary key constraint does not follow Prisma's convention (the conventional name would be `User_pk`):
+explain why you will probably want to propate the prod names into dev
```

Comment:
Capitalize the sentence here 😊

Reply:
I also dont understand this sentence, what are we saying here?

Reply:
I think its a note isnt it? We can remove it

Reply:
Yeah, I was not done yet ;-)

Reply:
oh sorry! 😬